### PR TITLE
move gun_islot to generic_factory

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary
Infrastructure "move gun_islot to generic_factory"

#### Purpose of change
part of #36851

#### Describe the solution
mostly some copy pasting and making sure i don't get any compile errors.

#### Testing
compile, then run the game. it'd be a good idea to check some guns.

### CODE REVIEW

File: src/item_factory.cpp [hunk 1]
```diff
@@ -1865,52 +1865,67 @@ void gun_variant_data::load( const JsonObject &jo )
...
-    assign( jo, "default_mods", slot.default_mods, strict );
-    assign( jo, "ups_charges", slot.ups_charges, strict, 0 );
-    assign( jo, "blackpowder_tolerance", slot.blackpowder_tolerance, strict, 0 );
-    assign( jo, "min_cycle_recoil", slot.min_cycle_recoil, strict, 0 );
-    assign( jo, "ammo_effects", slot.ammo_effects, strict );
-    assign( jo, "ammo_to_fire", slot.ammo_to_fire, strict, 1 );
-
-    optional( jo, false, "variants", slot.variants );
+    optional( jo, was_loaded, "ranged_damage", damage,
+              damage_instance( damage_type::NONE, -20, -20, -20, -20 ) );
```
Comment (anothersimulacrum) [inline src/item_factory.cpp:1875]: This argument in assign is not the default, it's the minimum value.

**Comment by KorGgenT** [comment]
looks like i'll have to write some special casing for damage_instance in proportional, unfortunately. or something.

**Comment by anothersimulacrum** [comment]
Add an `operator*=` to damage instances.

**Comment by KorGgenT** [comment]
i'll come back to this some other time.

**Pull request closed without merging**